dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcInvocation.java

```diff
@@ -397,7 +397,7 @@ public Map<String, String> getAttachments() {
 
     @Deprecated
     public void setAttachments(Map<String, String> attachments) {
-        this.attachments = attachments == null ? new HashMap<>() : new HashMap<>(attachments);
```

Comment:
'attachments' passed to setAttachments might haven't been synchronized, so it is a requirement of the setter to do that.

But I think it's ok to wrap the newly created map as synchronized.